### ISSUE TITLE
AI_ADDRCONFIG defined but leading to EAI_BADFLAGS

### ISSUE THREAD

**Issue by MedourMehdi** [freemint/mintlib#71]
Hi,

I've built gnutls_server binary - and It won't start returning me EAI_BADFLAGS.
I've fixed my issue by removing AI_ADDRCONFIG from the hints.flags.
The gnutls code looks like
```c
int
listen_socket (const char *name, int listen_port, int socktype)
{
  struct addrinfo hints, *res, *ptr;
  char portname[6];
  int s;
  int yes;
  listener_item *j = NULL;

  snprintf (portname, sizeof (portname), "%d", listen_port);
  memset (&hints, 0, sizeof (hints));
  hints.ai_socktype = socktype;
  hints.ai_flags = AI_PASSIVE
#ifndef __MINT__ /* Seems broken on Freemint */
#ifdef AI_ADDRCONFIG
    | AI_ADDRCONFIG
#endif

#endif
    ;

  if ((s = getaddrinfo (NULL, portname, &hints, &res)) != 0)
    {
      fprintf (stderr, "getaddrinfo() failed: %s\n", gai_strerror (s));
      return -1;
    }
````

I've seen in mintlib [code](https://github.com/freemint/mintlib/blob/b73c57604079ed9151c1b49105468253f550a0b1/socket/getaddrinfo.c#L117) that AI_ADDRCONFIG isn't handled by the function.

**Comment by th-otto** [comment]
Yes, i think  i stumbled already upon this in another library. Maybe better to remove the definition of AI_ADDRCONFIG from headers, or implement it somehow. But i'm also not sure what information it is supposed to return.

**Comment by MedourMehdi** [comment]
I guess the better would be to loop other interface != loopback then select the first ipaddr available - If not returning the loopback seems good to me?

**Comment by th-otto** [comment]
Looking at the implementation in NetBSD (https://github.com/NetBSD/src/blob/a0d5c31319d3cb7214a4d7e6925be6a41c5678e7/lib/libc/net/getaddrinfo.c#L528), it seems all it does is to filter out interfaces that don't support the requested address family. Given that all drivers only support IPv4, that shouldn't be hard to implement ;)

**Comment by MedourMehdi** [comment]
[getaddrinfo.diff.zip](https://github.com/user-attachments/files/16052338/getaddrinfo.diff.zip)
Don't really know how test it.
